eZ/Publish/Core/Persistence/Cache/TrashHandler.php

```diff
@@ -134,15 +138,19 @@ public function deleteTrashItem($trashedId)
         $trashed = $this->persistenceHandler->trashHandler()->loadTrashItem($trashedId);
 
         $reverseRelations = $this->persistenceHandler->contentHandler()->loadReverseRelations($trashed->contentId);
+
         $tags = array_map(function (Relation $relation) {
             return 'content-fields-' . $relation->sourceContentId;
         }, $reverseRelations);
 
         $return = $this->persistenceHandler->trashHandler()->deleteTrashItem($trashedId);
 
-        if (!empty($tags)) {
-            $this->cache->invalidateTags($tags);
-        }
+        $this->cache->invalidateTags([
+            'content-' . $return->contentId,
+            'content-fields-' . $return->contentId,
+            'location-' . $trashedId,
+            'location-path-' . $trashedId,
+        ] + $tags);
```

Comment:
This is not going to work, see what happens if:
```php
var_dump(['content-fields-1'] + ['content-fields-2']);
```
Additionally, if tests won't crash here, you'll need to add missing coverage :upside_down_face:

Reply:
Yup, you are right, I have fixed it also for trashSubtree now.